src/utils/daos.c

```diff
@@ -0,0 +1,870 @@
+			fprintf(stderr, "failed to connect to pool: %d\n", rc);
+			goto bad_pool_connect;
+		}
+
+		pinfo.pi_bits = DPI_ALL;
+		rc = daos_pool_query(pool, NULL, &pinfo, NULL, NULL);
+		if (rc != 0) {
+			fprintf(stderr, "pool query failed: %d\n", rc);
+			goto bad_pool_query;
+		}
+		D_PRINT("Pool "DF_UUIDF", ntarget=%u, disabled=%u\n",
+			DP_UUID(pinfo.pi_uuid), pinfo.pi_ntargets,
+			pinfo.pi_ndisabled);
+
+		D_PRINT("Pool space info:\n");
+		D_PRINT("- Target(VOS) count:%d\n", ps->ps_ntargets);
+		for (i = DAOS_MEDIA_SCM; i < DAOS_MEDIA_MAX; i++) {
+			D_PRINT("- %s:\n",
+				i == DAOS_MEDIA_SCM ? "SCM" : "NVMe");
+			D_PRINT("  Total size: "DF_U64"\n",
+				ps->ps_space.s_total[i]);
+			D_PRINT("  Free: "DF_U64", min:"DF_U64", max:"DF_U64", "
+				"mean:"DF_U64"\n", ps->ps_space.s_free[i],
+				ps->ps_free_min[i], ps->ps_free_max[i],
+				ps->ps_free_mean[i]);
+		}
+
+		if (rstat->rs_errno == 0) {
+			char	*sstr;
+
+			if (rstat->rs_version == 0)
+				sstr = "idle";
+			else if (rstat->rs_done)
+				sstr = "done";
+			else
+				sstr = "busy";
+
+			D_PRINT("Rebuild %s, "DF_U64" objs, "DF_U64" recs\n",
+				sstr, rstat->rs_obj_nr, rstat->rs_rec_nr);
+		} else {
+			D_PRINT("Rebuild failed, rc=%d, status=%d\n",
+				rc, rstat->rs_errno);
+		}
+	}
+
+	/* TODO implement the following:
+	 * op == POOL_LIST_CONTAINERS
+	 * op == POOL_STAT
+	 * op == POOL_GET_PROP
+	 * op == POOL_GET_ATTR
+	 * op == POOL_LIST_ATTRS
+	 */
+
+bad_pool_query:
+	/* Pool disconnect  in normal and error flows: preserve rc */
+	if (op == POOL_QUERY) {
+		rc2 = daos_pool_disconnect(pool, NULL);
+		if (rc2 != 0) {
+			fprintf(stderr, "Pool disconnect failed : %d\n", rc2);
+		}
+		if (rc == 0)
+			rc = rc2;
+	}
+bad_pool_connect:
+	return rc;
+}
+
+static int
+cont_op_hdlr(struct cmd_args_s *ap)
+{
+	daos_handle_t		pool;
+	daos_handle_t		coh;
+	daos_cont_info_t	cont_info;
+	int			rc;
+	int			rc2;
+	enum cont_op		op;
+	const int		RC_PRINT_HELP = 2;
+
+	assert(ap != NULL);
+	op = ap->c_op;
+
+	/* TODO: create functions per container op */
+
+	if (uuid_is_null(ap->cont_uuid)) {
+		fprintf(stderr, "valid cont uuid required\n");
+		fflush(stderr);
+		rc = RC_PRINT_HELP;
+		goto bad_opt_free_mdsrv;
+	}
+
+	/*
+	 * all cont operations require a pool handle, lets make a
+	 * pool connection
+	 */
+	rc = daos_pool_connect(ap->pool_uuid, ap->group, ap->mdsrv,
+			       DAOS_PC_RW, &pool,
+			       NULL /* info */, NULL /* ev */);
+	if (rc != 0) {
+		fprintf(stderr, "failed to connect to pool: %d\n", rc);
+		goto bad_pool_connect;
+	}
+
+	if (op == CONT_CREATE) {
+		/* TODO: add use case: create by pool UUID + path (UNS) */
+		rc = daos_cont_create(pool, ap->cont_uuid, NULL, NULL);
+		if (rc != 0) {
+			fprintf(stderr, "failed to create container: %d\n", rc);
+			goto bad_cont_create;
+		}
+		fprintf(stdout, "Successfully created container "DF_UUIDF"\n",
+			DP_UUID(ap->cont_uuid));
+	}
+
+	if (op != CONT_DESTROY) {
+		rc = daos_cont_open(pool, ap->cont_uuid, DAOS_COO_RW, &coh,
```

Comment:
why do you need to open/close the container in case of CREATE op?

Reply:
copy/paste from dcont. Will change the conditional to check for op != CREATE && op != DESTROY